src/engine.rs

```diff
@@ -15,17 +15,18 @@ use crate::consistency::ConsistencyChecker;
 use crate::env::{DefaultFileSystem, FileSystem};
 use crate::event_listener::EventListener;
 use crate::file_pipe_log::debug::LogItemReader;
-use crate::file_pipe_log::{
-    DefaultMachineFactory, FilePipeLog, FilePipeLogBuilder, RecoveryConfig,
-};
+use crate::file_pipe_log::{DefaultMachineFactory, FilePipeLog, FilePipeLogBuilder};
 use crate::log_batch::{Command, LogBatch, MessageExt};
 use crate::memtable::{EntryIndex, MemTableRecoverContextFactory, MemTables};
 use crate::metrics::*;
-use crate::pipe_log::{FileBlockHandle, FileId, LogQueue, PipeLog};
+use crate::pipe_log::{FileBlockHandle, FileId, LogQueue, PipeLog, Signature};
 use crate::purge::{PurgeHook, PurgeManager};
 use crate::write_barrier::{WriteBarrier, Writer};
 use crate::{Error, GlobalStats, Result};
 
+#[cfg(feature = "scripting")]
+use crate::file_pipe_log::RecoveryConfig;
```

Comment:
Why?

Reply:
Here, the reason why i did this is for keeping the code style coincident with the calling function.
The `RecoveryConfig` is only called in the implementation of `fn unsafe_repair_with_file_system`, which is restricted with `feature = "scripting"`.  So i did it here.

Reply:
Oh, I prefer not. I think you can move the `use ...` inside the function.

Reply:
Oh, yep, it's a preferrable choice for it. I just forget it.
Got it.